krod/agents/agent_manager.py

```
@@ -0,0 +1,400 @@
+                # default to ResearchAgent on error
+                return [a for a in self.agents if a.get("name") == "ResearchAgent" and a.get("enabled", True)]
+        except Exception as e:
+            self.logger.error(f"Error selecting agents: {str(e)}", exc_info=True)
+            # default to ResearchAgent on error
+            return [a for a in self.agents if a.get("name") == "ResearchAgent" and a.get("enabled", True)]
+    
+    async def _invoke_agent(
+        self, 
+        agent_config: Dict, 
+        query: str, 
+        context: AgentContext
```

Flake8 (7.2.0)

[error] undefined name 'AgentContext'

(F821)